irc_channel.c

```diff
@@ -760,7 +760,7 @@ static gboolean control_channel_invite(irc_channel_t *ic, irc_user_t *iu)
 	}
 
 	bu->ic->acc->prpl->add_buddy(bu->ic, bu->handle,
-	                             icc->group ? icc->group->name : NULL);
+	                             icc->groups ? ((struct bee_group *)icc->groups->data)->name : NULL);
```

Comment:
You could move this to a different line, and assign `->data` to a variable of the intended type to save yourself the cast. Same below in the kick function